GUI/window.py

```diff
@@ -211,21 +364,35 @@ def run_micro_instructions_step(self, instance):
             else:
                 self.step_index += 1
                 if self.first_inst:
-                    self.inst_table.get_data(self.micro_sim.index, self.header, self.micro_sim.disassembled_instruction())
+                    self.inst_table.get_data(self.micro_sim.index, self.header,
+                                             self.micro_sim.disassembled_instruction())
                     self.first_inst = False
+                    self.update_grid()
                 else:
 
                     self.micro_sim.run_micro_instructions_step(self.step_index)
                     self.reg_table.get_data()
                     self.mem_table.data_list.clear()
                     self.mem_table.get_data()
-                    self.inst_table.get_data(self.micro_sim.index, self.header, self.micro_sim.disassembled_instruction())
+                    self.inst_table.get_data(self.micro_sim.index, self.header,
+                                             self.micro_sim.disassembled_instruction())
+                    self.light.change_color(self.micro_sim.traffic_lights_binary())
+                    self.update_grid()
 
                 toast('Runnin instruction in step-by-step mode. Step ' + str(self.step_index) + ' is running')
                 for i in self.micro_sim.micro_instructions:
                     if i != 'NOP':
                         print(i)
 
+    def update_grid(self):
```

Comment:
Consider renaming this method. `RunWindow` has various widgets so method should be clear on what widget or in this case I/O it is trying to update the grid for. Another option would be to move this method the `ASCIIGrid`